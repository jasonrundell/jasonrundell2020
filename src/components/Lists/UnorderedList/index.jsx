import React from 'react'
import PropTypes from 'prop-types'

import styles from './UnorderedList.module.scss'

const UnorderedList = ({ items, children }) => {
  return (
    <ul className={styles.root}>
      {items.map((item, index) => (
        <li className={styles.listItem} key={index}>
          {item}{children}
        </li>
      ))}
    </ul>
  )
}

UnorderedList.propTypes = {
  items: PropTypes.array.isRequired,
  children: PropTypes.any
}

export default UnorderedList
