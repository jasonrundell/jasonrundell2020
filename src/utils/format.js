/**
 *  Letter case conversion functions
 *  [Letter case](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)
 *
 *  Converts only letter case-separated or hyphen-delimited strings
 */

/**
 *  Converts hyphen-delimited `string` to camel case
 *
 *  @example
 *  'foo-bar' => 'fooBar'
 *  'foo--bar' => 'fooBar'
 *
 *  @param {string} string
 */
function toCamelCase(string) {
  return string.replace(/(-+)([a-z])/g, ($0, $1, $2) => $2.toUpperCase())
}

/**
 *  Converts hyphen-delimited `string` to pascal case (aka upper camel case)
 *
 *  @example
 *  'foo-bar' => 'FooBar'
 *  'foo--bar' => 'FooBar'
 *
 *  @param {string} string
 */
function toPascalCase(string) {
  return toCamelCase(string.replace(/^[a-z]/, $0 => $0.toUpperCase()))
}

export { toCamelCase, toPascalCase }
