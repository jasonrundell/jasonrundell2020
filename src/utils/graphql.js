export function edgesToObject(edgesData) {
  const nodes = edgesData.map(item => {
    return item.node
  })
  const finalObject = nodes.map(item => {
    return item
  })

  return finalObject
}
