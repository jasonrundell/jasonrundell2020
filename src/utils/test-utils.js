export function catchWarnings() {
  const originalConsoleError = global.console.error
  beforeEach(() => {
    /**
     * I Want to know if values passed to props fail the type check
     * */
    global.console.error = (...args) => {
      const propTypeFailures = [/Warning:/] // e.g. [/Invalid prop/, /Failed prop type/]

      if (propTypeFailures.some(p => p.test(args[0]))) {
        throw new Error(args[0])
      }

      originalConsoleError(...args)
    }
  })
}

/**
 * Mix an match help from:
 * https://daveceddia.com/testing-for-proptypes-errors/
 * https://www.jackfranklin.co.uk/blog/failing-tests-on-react-proptypes/
 */
