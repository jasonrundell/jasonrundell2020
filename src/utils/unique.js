const onlyUnique = (value, index, self) => {
  return self.indexOf(value) === index
}

const generatePassword = ({
  passwordLength = 16,
  charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_-=~`',
}) => {
  let retVal = ''
  for (var i = 0, n = charset.length; i < passwordLength; ++i) {
    retVal += charset.charAt(Math.floor(Math.random() * n))
  }
  return retVal
}

export { onlyUnique, generatePassword }
