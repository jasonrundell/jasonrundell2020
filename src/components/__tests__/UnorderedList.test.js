import React from 'react'
import { unmountComponentAtNode } from 'react-dom'
import renderer from 'react-test-renderer'
import { catchWarnings } from '../../utils/test-utils'

import UnorderedList from '../Lists/UnorderedList'

catchWarnings()

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

const listData = ['one', 'two', 'three']

describe('UnorderedList component', () => {
  it('UnorderedList renders correctly', () => {
    const tree = renderer
      .create(
        <div>
          <UnorderedList list={listData} />
          <UnorderedList list={listData} isHorizontal={true} />
          <UnorderedList list={listData}>
            <UnorderedList list={listData} />
          </UnorderedList>
        </div>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
