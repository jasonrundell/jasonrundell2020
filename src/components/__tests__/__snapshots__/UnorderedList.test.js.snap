// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnorderedList component UnorderedList renders correctly 1`] = `
<div>
  <ul
    className="root"
  >
    <li
      className="listItem"
    >
      one
    </li>
    <li
      className="listItem"
    >
      two
    </li>
    <li
      className="listItem"
    >
      three
    </li>
  </ul>
  <ul
    className="root horizontal"
  >
    <li
      className="listItem"
    >
      one
    </li>
    <li
      className="listItem"
    >
      two
    </li>
    <li
      className="listItem"
    >
      three
    </li>
  </ul>
  <ul
    className="root"
  >
    <li
      className="listItem"
    >
      one
      <ul
        className="root"
      >
        <li
          className="listItem"
        >
          one
        </li>
        <li
          className="listItem"
        >
          two
        </li>
        <li
          className="listItem"
        >
          three
        </li>
      </ul>
    </li>
    <li
      className="listItem"
    >
      two
      <ul
        className="root"
      >
        <li
          className="listItem"
        >
          one
        </li>
        <li
          className="listItem"
        >
          two
        </li>
        <li
          className="listItem"
        >
          three
        </li>
      </ul>
    </li>
    <li
      className="listItem"
    >
      three
      <ul
        className="root"
      >
        <li
          className="listItem"
        >
          one
        </li>
        <li
          className="listItem"
        >
          two
        </li>
        <li
          className="listItem"
        >
          three
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
